### ISSUE TITLE
RUF076: should not fire in conftest.py by default (autouse is the standard pattern there)

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [astral-sh/ruff#26178]
## Summary

`RUF076` (pytest-fixture-autouse) discourages `autouse=True` in pytest fixtures, but fires in `conftest.py` files by default. This is problematic because `conftest.py` is precisely where `autouse=True` fixtures are most appropriate and widely used — it's the standard pytest pattern for applying shared setup across a test directory.

The rule's own documentation acknowledges this tension:

> This is a pedantic rule that restricts a valid pytest pattern.

And even suggests a workaround to *only* apply it in conftest.py:

```toml
[tool.ruff.lint.per-file-ignores]
"!**/conftest.py" = ["RUF076"]
```

But this inverts the correct default. The rule should **not** fire in `conftest.py` by default — or at minimum, there should be a configuration option to scope it to non-conftest files only. Autouse fixtures in conftest.py are the documented, recommended pytest approach for directory-scoped setup (e.g., environment variables, database cleanup, logging config).

## Reproduction

```python
# conftest.py
import pytest

@pytest.fixture(autouse=True)
def setup_test_env() -> None:
    """Standard conftest autouse fixture for test environment setup."""
    ...
```

```toml
[tool.ruff]
lint.extend-select = ["RUF"]
lint.preview = true
```

```
$ ruff check conftest.py
RUF076: Avoid using `autouse=True` in `pytest.fixture` decorators
```

## Expected behavior

Either:
1. `RUF076` should not fire in `conftest.py` by default (since that's where autouse is most appropriate), OR
2. The default should be inverted: only fire in conftest.py if explicitly configured, and fire in test files by default (where autouse is actually problematic)

## Ruff version

```
ruff 0.15.18
```

## Context

The original rule request (#12491) specifically called out conftest.py as problematic, but in practice, `autouse` in conftest is the standard pytest pattern documented at:
- https://docs.pytest.org/en/stable/how-to/fixtures.html#autouse-fixtures-fixtures-you-don-t-have-to-request
- https://docs.pytest.org/en/stable/how-to/fixtures.html#conftest-py-sharing-fixtures-across-tests

The rule is useful for discouraging autouse in individual test files (where it's truly confusing), but not in conftest.py where it's expected behavior.

**Comment by tjkuson** [comment]
> widely used

Here's some supporting data I posted on the original issue: [ref](https://grep.app/search?f.path.pattern=conftest.py&q=autouse%3DTrue)

I'd also add that RUF076 diagnostics in conftest.py files are uniquely unactionable in that there isn't a viable alternative to using them (I think it's unreasonable to ask users to add `pytest.mark.usefixtures` to every test in a directory).

**Comment by ntBre** [comment]
Thanks for the nice write-up! I'm still hesitant to add a special-purpose configuration option to make the rule apply or not apply to conftest.py files when `per-file-ignores` is an existing mechanism for the same purpose. However, I think it could make sense for the rule documentation to be much less opinionated on conftest.py and also show examples/discuss enabling the rule only outside conftest.py, as you say.

As someone who hasn't used that much pytest, I found the current description compelling, as code being injected into my tests from another file sounded particularly confusing. But if that's the typical pytest usage we should include it in the docs.

**Comment by RonnyPfannschmidt** [comment]
autouse fixtures are a key utility in both plugins and testsuites - they very important - discouraging the min this way as the rule does atm is very problematic from my pov

**Comment by RonnyPfannschmidt** [comment]
basically if this rule stays as is - i'll have to do at least a hundred pull reqeusts to ignore it just to cover whats related to my work - iff i count in coworkers we are looking at easily greater than 2000

**Comment by ntBre** [comment]
> basically if this rule stays as is - i'll have to do at least a hundred pull reqeusts to ignore it just to cover whats related to my work - iff i count in coworkers we are looking at easily greater than 2000

Is this because it was released as a `RUF` rule? Our intention was not for this rule to be enabled by default, but if you have preview and the RUF rules enabled, then I guess it will have started firing immediately. We might need to revert this until we have a better category for it.

**Comment by RonnyPfannschmidt** [comment]
We opt in to a good number as usually most rules are somewhere between good maybe arguable 

This is the first nope I hit myself

**Comment by flying-sheep** [comment]
Almost the same for me, I even have a project with both `ALL` and `--preview` on. Other nopes for me are basically only `B019` (banning `functools.cache`, but I think it could be useful for some people), and a few of the `S` rules (which I think are self-aware about it, e.g. [`S603`](https://docs.astral.sh/ruff/rules/subprocess-without-shell-equals-true/) seems to exist to have a `noqa`/`nosec` comment act as a “beware” marker, not to actually ban `subprocess.run` completely)
